blockpoller/poller.go

```diff
@@ -35,6 +36,8 @@ type BlockPoller struct {
 	forkDB       *forkable.ForkDB
 
 	logger *zap.Logger
+
+	optimisticallyPolledBlocks map[uint64]*BlockItem
```

Comment:
We should add a comment here to explain why it's safe/desired behavior to use the block number as a key here. Myself by quickly reading the code I cannot for sure says why.